ci-checks.sh

```diff
@@ -6,7 +6,8 @@ set -eux
 
 failed=0
 
-if ! git ls-files | xargs codespell -q 3 -I .codespell-whitelist; then
+# spell-checks only language files to avoid spell-checking checksums
+if ! git ls-files '*.sh' '*.go' '*.md' | xargs codespell -q 3 -I .codespell-whitelist; then
```

Comment:
what about the opposite, exclude `go.sum` and `go.mod`? This way any new files introduced are checked by default (which may or may not be a good thing ;) )

Reply:
This was a conscious decision: The number of file extensions where it is valuable to spell-check code for is finite, whereas the number of file extensions that contain other data is unlimited:
 
* dockerignore
* editorconfig
* envrc
* gitignore
* idx
* json
* lock
* mk
* mod
* nix
* pack
* proto
* sample
* sum
* tmpl
* toml
* txt
* yaml
* yml

One could argue for adding some of these extensions - but a whitelist of code extensions will be far easier to maintain than a blacklist of data files.

Reply:
:+1: makes sense